app/controllers/questions_controller.rb

```diff
@@ -19,9 +23,111 @@ def show
       gon.solution = solution
     end
 
+    if !(current_user.show_solutions)
```

Comment:
Odporucam skor opacne poradie podmienok - tak, aby nebol priamo v podmienke zapor (na to sa pouziva v Railse `unless`, ale iba ak nepotrebujes `else` vetvu).
```
if current_user.show_solutions
 kod, ktory sa vykona, ak chceme zobrazovat riesenia pouzivatela
else
 kod, ktory sa vykona, ak nechceme zobrazovat riesenia pouzivatela
end
```